src/components/Analytics/AnalyticsMenu.tsx

```diff
@@ -48,21 +68,23 @@ const AnalyticsMenu = ({ headerElem }) => {
 
   // The charts to display, must contain 'link' and 'label' properties so that they can easily be passed
   // to AutoScrollableMenu component
-  const [charts, setCharts] = useState(generalCharts(t));
+  const [charts, setCharts] = useState<Chart[]>(generalCharts(t));
   const chartRefs = useRef({});
   const chartContainerRef = useRef(null);
-  const [stackElem, setStackElem] = useState(null);
-  const [sideMenuElem, setSideMenuElem] = useState(null);
+  const [stackElem, setStackElem] = useState<HTMLElement | null>(null);
+  const [sideMenuElem, setSideMenuElem] = useState<HTMLElement | null>(null);
   const chartTabs = useRef(null);
-  const maxResultViewHeight = useMaxAvailableHeightInWindow(headerElem.current);
+  const maxResultViewHeight = useMaxAvailableHeightInWindow(
+    headerElem?.current ?? undefined
+  );
   const maxHeightScrollableMenu = useMaxAvailableHeightWithParentHeight(
     maxResultViewHeight,
-    chartTabs.current
+    chartTabs?.current ?? undefined
```

Comment:
won't the first param will be undefined anyway? Or is it to avoid `null`?

Reply:
It is to avoid null, but you're right the chartTabs is never null, it is current.